File: 1346. Check If N and Its Double Exist/Solution.cs
public class Solution {
    public bool CheckIfExist(int[] arr) {

        // Corner case 0*2 = 0
        if (arr.Count(n => n == 0) > 1) return true;
        
        HashSet<int> uniqueNumbers = new HashSet<int>(arr);

        foreach (int num in arr) {
            if (uniqueNumbers.Contains(num*2) && num != 0) {
                return true;
            }
        }
        return false;
    }
}

File: 1346. Check If N and Its Double Exist/solution.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        
        # Corner case 0*2 = 0
        if arr.count(0) > 1:
            return True
        
        unique_numbers = set(arr)

        for num in unique_numbers:
            if num*2 in unique_numbers and num != 0:
                return True

        return False
